python/examples/tools/mcp_agent.py

```diff
@@ -0,0 +1,132 @@
+import asyncio
+import logging
+import os
+import sys
+import traceback
+from typing import Any
+
+from dotenv import load_dotenv
+from mcp import ClientSession, StdioServerParameters
+from mcp.client.stdio import stdio_client
+
+from beeai_framework import Tool
+from beeai_framework.agents.bee.agent import BeeAgent
+from beeai_framework.agents.types import BeeAgentExecutionConfig
+from beeai_framework.backend.chat import ChatModel, ChatModelParameters
+from beeai_framework.emitter.emitter import Emitter, EventMeta
+from beeai_framework.emitter.types import EmitterOptions
+from beeai_framework.errors import FrameworkError
+from beeai_framework.memory.token_memory import TokenMemory
+from beeai_framework.tools.mcp_tools import MCPTool
+from beeai_framework.utils.custom_logger import BeeLogger
+from examples.helpers.io import ConsoleReader
+
+# Load environment variables
+load_dotenv()
+
+reader = ConsoleReader()
+
+# Configure logging - using DEBUG instead of trace
+logger = BeeLogger("app", level=logging.DEBUG)
+
+# Create server parameters for stdio connection
+server_params = StdioServerParameters(
+    command="npx",
+    args=["-y", "@modelcontextprotocol/server-slack"],
+    env={
+        "SLACK_BOT_TOKEN": "",
+        "SLACK_TEAM_ID": "",
+        "PATH": os.getenv("PATH", default=""),
+    },
+)
+
+
+async def slack_tool() -> MCPTool:
+    async with stdio_client(server_params) as (read, write), ClientSession(read, write) as session:
+        await session.initialize()
+        # Discover Slack tools via MCP client
+        slacktools = await MCPTool.from_client(session, server_params)
+        filter_tool = filter(lambda tool: tool.name == "slack_post_message", slacktools)
+        slack = list(filter_tool)
+        return slack[0]
+
+
+async def create_agent() -> BeeAgent:
+    """Create and configure the agent with tools and LLM"""
+
+    # Other models to try:
+    # "llama3.1"
+    # "granite3.1-dense"
+    # "deepseek-r1"
+    # ensure the model is pulled before running
+    llm = ChatModel.from_name(
+        "ollama:llama3.1",
+        ChatModelParameters(temperature=0),
+    )
+
+    # Configure tools
+    tools: list[Tool] = [await slack_tool()]
+
+    # Add code interpreter tool if URL is configured
+    code_interpreter_url = os.getenv("CODE_INTERPRETER_URL")
+    if code_interpreter_url:
+        # Note: Python tool implementation would go here
+        pass
```

Comment:
can this be replaced with a single `TODO` line to include code interpreter once implemented/avaiable?

Reply:
I would make the example as simple as possible. Remove all unnecessary config/tools and so on.